src/core/worker.js

```diff
@@ -355,6 +355,16 @@ var WorkerMessageHandler = {
     handler.on('GetDocRequest', function wphSetupDoc(data) {
       return WorkerMessageHandler.createDocumentHandler(data, port);
     });
+// what listener should I add here? Please suggest.
+    handler.on('', function workerCoverage() {
```

Comment:
I suggest `GetCoverageData`. This is a new event name, it is your task to send a message at a meaningful moment (e.g. after all tests have concluded).

You could add an `if` check to see whether test coverage is enabled (check whether `globalScope.__coverage__` is truthy), and if so, send the `GetCoverageData` message and wait for its completion by pusing the promise in the `waitOn` list.

https://github.com/mozilla/pdf.js/blob/f774abc8d335712fb14ebc6dc262eb389363607b/src/display/api.js#L1541-L1544